app/views/core/VueComponentView.js

```diff
@@ -30,19 +31,32 @@ module.exports = class VueComponentView extends RootView {
     })
   }
 
+  setState (state = {})  {
+    for (const key of Object.keys(state)) {
+      Vue.set(
+        this.props,
+        key,
+        state[key]
+      )
+    }
+  }
+
   afterRender() {
     if (this.vueComponent) {
       this.$el.find('#site-content-area').replaceWith(this.vueComponent.$el)
-    }
-    else{
+    } else {
       this.vueComponent = this.buildVueComponent()
       window.rootComponent = this.vueComponent
     }
 
+    this.propsData = this.vueComponent.$props
+
     super.afterRender()
   }
 
-  destroy() {
+  destroy () {
+    super.destroy()
```

Comment:
Note: we should add this on master if this PR doesn't eventually get improved / merged into master.